### PR TITLE
:scissors: jQuery for Rails Guides

### PR DESCRIPTION
**Pull request by yhirano55**
### Summary

I feel javascripts are so simple implementation in Rails Guides. So it doesn't need for them to use jQuery. And I've rewritten with Vanilla JavaScript.

* Re-wrote with Vanilla JS (Use `querySelector` instead of `$()`)
* Removed jQuery file
* Use strict mode
* Use IIFE

Confirmed browsers: 

```
- Chrome 65.0.3325.181
- Safari 11.1 (13605.1.33.1.2)
- Firefox 59.0.2
```

/cc @guilleiguaran @javan 

---

##### Before

```javascript
    $("#guidesMenu").on("click", function(e) {
      $("#guides").toggle();
      return false;
    });
```

##### After

```javascript
  document.addEventListener("DOMContentLoaded", function() {
    var $guidesMenu = document.getElementById("guidesMenu");
    var $guides     = document.getElementById("guides");

    $guidesMenu.addEventListener("click", function(e) {
      e.preventDefault();
      $guides.classList.toggle("visible");
    });
```

---

##### Before

```javascript
    var currentGuidePath = window.location.pathname;
    var currentGuide = currentGuidePath.substring(currentGuidePath.lastIndexOf("/")+1);
    $(".guides-index-small").
      on("change", "select", guidesIndex.navigate).
      selectGuide(currentGuide);
    $(document).on("click", ".more-info-button", function(e){
      e.stopPropagation();
      if ($(".more-info-links").is(":visible")) {
        $(".more-info-links").addClass("s-hidden").unwrap();
      } else {
        $(".more-info-links").wrap("<div class='more-info-container'></div>").removeClass("s-hidden");
      }
    });
    $(document).on("click", function(e){
      e.stopPropagation();
      var $button = $(".more-info-button");
      var element;

      // Cross browser find the element that had the event
      if (e.target) element = e.target;
      else if (e.srcElement) element = e.srcElement;

      // Defeat the older Safari bug:
      // http://www.quirksmode.org/js/events_properties.html
      if (element.nodeType === 3) element = element.parentNode;

      var $element = $(element);

      var $container = $element.parents(".more-info-container");

      // We've captured a click outside the popup
      if($container.length === 0){
        $container = $button.next(".more-info-container");
        $container.find(".more-info-links").addClass("s-hidden").unwrap();
      }
    });
```

##### After

```javascript
    var $guidesIndexItem   = document.querySelector("select.guides-index-item");
    var currentGuidePath   = window.location.pathname;
    $guidesIndexItem.value = currentGuidePath.substring(currentGuidePath.lastIndexOf("/") + 1);

    $guidesIndexItem.addEventListener("change", function(e) {
      window.location = e.target.value;
    });

    var $moreInfoButton = document.querySelector(".more-info-button");
    var $moreInfoLinks  = document.querySelector(".more-info-links");

    $moreInfoButton.addEventListener("click", function(e) {
      e.preventDefault();

      if ($moreInfoLinks.classList.contains("s-hidden")) {
        wrap($moreInfoLinks, createElement("div", "more-info-container"));
        $moreInfoLinks.classList.remove("s-hidden");
      } else {
        $moreInfoLinks.classList.add("s-hidden");
        unwrap($moreInfoLinks);
      }
    });
```

---

##### Before

See the file directly.

##### After

See the file directly.

---

##### Before

image

```html
  <script type="text/javascript" src="javascripts/syntaxhighlighter.js"></script>
  <script type="text/javascript">
    syntaxhighlighterConfig = {
      autoLinks: false,
    };
    $(guidesIndex.bind);
  </script>
```

##### After

```javascript
(function() {
  "use strict";
  window.syntaxhighlighterConfig = { autoLinks: false };
```

### CODE REVIEW

**Comment by guilleiguaran** [comment]
We will need to confirm compatibility with IE before of merging this

**Comment by saiqulhaq** [comment]
it seems work with IE10+

**Comment by guilleiguaran** [comment]
sounds good 😊

File: guides/assets/javascripts/responsive-tables.js [hunk 1]
```diff
@@ -1,43 +1,43 @@
-$(document).ready(function() {
+(function() {
+  "use strict";
   var switched = false;
-  $("table").not(".syntaxhighlighter").addClass("responsive");
+
+  document.querySelectorAll(":not(.syntaxhighlighter)>table").forEach(function(element) {
```
Comment (javan) [inline guides/assets/javascripts/responsive-tables.js]: [`NodeList.prototype.forEach()`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach) isn't supported in Internet Explorer and some older browser versions. You'll need to slice these into an array first or use a `for` loop to iterate.

File: guides/assets/javascripts/guides.js [hunk 2]
```diff
@@ -1,53 +1,53 @@
...
-      }
-    });
-    $("#guidesMenu").on("click", function(e) {
-      $("#guides").toggle();
-      return false;
+(function() {
+  "use strict";
+  window.syntaxhighlighterConfig = { autoLinks: false };
+
+  this.wrap = function(elem, wrapper) {
```
Comment (javan) [inline guides/assets/javascripts/guides.js]: `this` is `window` here. Are you intending to make these helpers globally available?

File: guides/assets/javascripts/guides.js [hunk 3]
```diff
@@ -1,53 +1,53 @@
...
+  }
+
+  this.createElement = function(tagName, className) {
+    var elem = document.createElement(tagName);
+    elem.classList.add(className);
+    return elem;
+  }
+
+  document.addEventListener("DOMContentLoaded", function() {
+    var $guidesMenu = document.getElementById("guidesMenu");
```
Comment (javan) [inline guides/assets/javascripts/guides.js]: `$` in variable names is generally used to indicate that they're jQuery collections / objects.

**Pull request merged**